### ISSUE TITLE
ArrayBuilder.generic.addAll is broken on 2.13

### ISSUE THREAD

**Issue by sjrd** [scala-js/scala-js#3705]
https://github.com/scala-js/scala-js/blob/2e2e2bc0c0b4984a5ea646a9ca8b391f10f123fd/scalalib/overrides-2.13/scala/collection/mutable/ArrayBuilder.scala#L115-L120

The implementation passes `length` as second argument to `slice`, but that argument is an `until`. It should give `offset + length` to be compatible.

Failing test case:
```scala
    val b = mutable.ArrayBuilder.make[Int]
    val arr = Array[Int](1, 2, 3, 4, 5)
    b.addAll(arr, 3, 2)
    assertArrayEquals(Array[Int](4, 5), b.result())
```

This indirectly causes the upstream JUnit test `scala/collection/ArrayOpsTest.scala` to fail.

Note: the call to `ensureSize` is completely useless and should be removed while we're there.

**Comment by sjrd** [comment]
Fixed implementation:
```scala
    /** Add a slice of an array */
    override def addAll(xs: Array[_ <: T], offset: Int, length: Int): this.type = {
      val end = offset + length
      var i = offset
      while (i < end) {
        this += xs(i)
        i += 1
      }
      this
    }
```

**Comment by sjrd** [comment]
If someone wants to tackle this:

Fixing this should be done in the 0.6.x branch.

Tests for this issue should go to [`ArrayBuilderTest`](https://github.com/scala-js/scala-js/blob/0.6.x/test-suite/shared/src/test/scala/org/scalajs/testsuite/scalalib/ArrayBuilderTest.scala). Since `addAll` did not exist in earlier versions of Scala, you'll need to use some implicit magic similar to what's done at https://github.com/scala-js/scala-js/blob/2e2e2bc0c0b4984a5ea646a9ca8b391f10f123fd/test-suite/shared/src/test/require-sam/org/scalajs/testsuite/compiler/SAMTest.scala#L113-L121

**Comment by sjrd** [comment]
Fixed in 3f7b83b19176aac85a936801d7e57b9176ede0d1.
